modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderLocalServiceImpl.java

```diff
@@ -1257,6 +1298,33 @@ public CommerceOrder updateBillingAddress(
 			serviceContext, subtype, zip);
 	}
 
+	@Indexable(type = IndexableType.REINDEX)
+	@Override
+	public CommerceOrder updateCommerceOrder(CommerceOrder commerceOrder) {
+		try {
+			CommerceChannel commerceChannel =
+				_commerceChannelLocalService.getCommerceChannelByGroupId(
+					commerceOrder.getGroupId());
+
+			User user = _userLocalService.getUser(commerceOrder.getUserId());
+
+			_setAccountDefaultBillingAddress(commerceChannel, commerceOrder);
+			_setAccountDefaultShippingAddress(commerceChannel, commerceOrder);
+			_setAccountDefaultPaymentIntegrationKey(
+				commerceChannel, commerceOrder);
+			_setAccountDefaultShippingOption(
+				commerceChannel, commerceOrder, user);
+			_setAccountDefaultDeliveryTerm(
+				commerceChannel, commerceOrder, user);
+			_setAccountDefaultPaymentTerm(commerceChannel, commerceOrder, user);
```

Comment:
setting of commercePaymentMethodKey in setAccountDefaultPaymentIntegrationKey needs to come before setting of paymentCommerceTermEntryId in _setAccountDefaultPaymentTerm and setting of `deliveryCommerceTermEntryId` in `_setAccountDefaultDeliveryTerm`

Reply:
`_setAccountDefaultShippingOption` was intentionally placed after setter of the billingAddress and shippingAddress because we depend on the two values for the rest of the logic.